control/statesp.py

```diff
@@ -612,14 +612,20 @@ def dcgain(self):
             at the origin
         """
         try:
-            gain = np.asarray(self.D -
-                              self.C.dot(np.linalg.solve(self.A, self.B)))
+            if self.isctime():
+                gain = np.asarray(self.D -
+                                    self.C.dot(np.linalg.solve(self.A, self.B)))
+            else:
+                gain = self.horner(1)
         except LinAlgError:
-            # zero eigenvalue: singular matrix
-            return np.nan
+            # eigenvalue at DC
+            # TODO: should this be of shape (outputs,inputs) ?
+            # TODO: inf rather than nan?
+            # TODO: could one not do a bit better, maybe by finding a least squares solution?
```

Comment:
I think that these questions are good for discussion, but doing so in an issue or on the mailing list would be better than new TODO comments here.